### PR TITLE
KFSPTS-9545: Control by KFS system parameter PMW_INTEGRATION_IS_ACTIVE_IND the tracking of Approve/Disapprove/Cancel actions of PVEN for new domestic DV vendors whose data orignated from PMW or AVF or hand keyed.

### PR DESCRIPTION
**Pull request by nkimble**
cu-kfs PR#541 and nonprod-sql PR#300 must be merged together due to dependencies. 
This code tracks the asynchronous approve/disapprove/cancel actions a KFS user may perform on an enroute PVEN for a new domestic DV vendor. The data could have been entered via PMW, AVF, or hand keyed.

### CODE REVIEW

File: src/main/resources/edu/cornell/kfs/pmw/cu-ojb-pmw.xml [hunk 1]
```diff
@@ -19,6 +19,7 @@
 <descriptor-repository version="1.0">
 
 <class-descriptor class="edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksVendor" table="CU_PMW_VENDOR_T">
+    <field-descriptor name="id" column="ID" jdbc-type="INTEGER" primarykey="true" sequence-name="CU_PMW_VNDR_SEQ" autoincrement="true" />
```
Comment (db483) [inline src/main/resources/edu/cornell/kfs/pmw/cu-ojb-pmw.xml]: I did not see SQL for creating the CU_PMW_VNDR_SEQ, does it already exist?

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java [hunk 2]
```diff
@@ -107,6 +128,181 @@ public Person getSystemUser() {
...
+        return sb.toString();
+    }
+    
+    private boolean vendorNameIsBlank(VendorDetail vendorDetail) {
+        return (StringUtils.isBlank(vendorDetail.getVendorName()));
+    }
+
+    private void populateRemitAddress(PaymentWorksVendor pmwVendor, VendorDetail vendorDetail) {
+        List<VendorAddress> remitAddresses = findAllActiveRemitAddresses(vendorDetail.getVendorAddresses());
+        if (remitAddresses.size() == 1) {
```
Comment (db483) [inline src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: do we need to check for nulls here or does this always return a list?

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java [hunk 3]
```diff
@@ -107,6 +128,181 @@ public Person getSystemUser() {
...
+    private List<VendorAddress> findAllActiveRemitAddresses(List<VendorAddress> addresses) {
+        return (addresses.stream()
+                         .filter(address -> address.getVendorAddressTypeCode().equalsIgnoreCase(VendorConstants.AddressTypes.REMIT))
+                         .filter(address -> address.isActive())
+                         .collect(Collectors.toList()));
+    }
+    
+    private void populateContactEmail(PaymentWorksVendor pmwVendor, VendorDetail vendorDetail) {
+        List<VendorContact> vendorInformationContacts = findAllActiveVendorInformationContacts(vendorDetail.getVendorContacts());
+        if (vendorInformationContacts.size() == 1) {
```
Comment (db483) [inline src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: do we need to check for nulls here?

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java [hunk 4]
```diff
@@ -107,6 +128,181 @@ public Person getSystemUser() {
...
+    @Override
+    public void registerKfsPvenApprovalForKfsEditedVendor(String kfsVendorDocumentNumber, VendorDetail vendorDetail) {
+        LOG.info("registerKfsPvenApprovalForKfsEditedVendor: Approval for Edited Vendor with KFS document number '" + kfsVendorDocumentNumber + "'  NOT inserted into PaymentWorks table. PMW Edit functionality not present yet.");
+    }
+
+    private PaymentWorksVendor getPaymentWorksVendorByDocumentNumber(String documentNumber) {
+        Map criteria = new HashMap();
+        criteria.put(PaymentWorksPropertiesConstants.PaymentWorksVendor.KFS_VENDOR_DOCUMENT_NUMBER, documentNumber);
+        Collection<PaymentWorksVendor> newVendors = getBusinessObjectService().findMatching(PaymentWorksVendor.class, criteria);
+        if (newVendors.isEmpty()) {
```
Comment (db483) [inline src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: do we need to check for null?

File: src/main/resources/edu/cornell/kfs/pmw/cu-ojb-pmw.xml (hunk 1 above)
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/pmw/cu-ojb-pmw.xml]: The sequence already exists and was previously created by the archived SQL in [KFSPTS-9246-staging-table.sql]

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java (hunk 4 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: Adding as suggested.

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java (hunk 2 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: Adding as suggested

File: src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java (hunk 3 above)
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksBatchUtilityServiceImpl.java]: Adding as suggested

File: src/main/java/edu/cornell/kfs/pmw/batch/businessobject/PaymentWorksIsoCountryToFipsCountryAssociation.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    private static final long serialVersionUID = -229089609514914655L;
+    
+    private Integer id;
+    private String isoCountryCode;
+    private String fipsCountryCode;
+   
+    public Integer getId() {
+        return id;
+    }
+    public void setId(Integer id) {
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/pmw/batch/businessobject/PaymentWorksIsoCountryToFipsCountryAssociation.java]: Fixing

**Pull request merged**